include/range/v3/utility/functional.hpp

```diff
@@ -181,6 +181,17 @@ namespace ranges
           : coerce<T>
         {};
 
+        struct dereference_fn
+        {
+            template<typename I>
+            constexpr auto operator()(I &i) const
```

Comment:
I don't want to add more functions that return possibly-dangling references obtained from temporary iterators. (https://github.com/ericniebler/stl2/issues/156)